eslint_rules/no-undefined-identifier.js

```diff
@@ -0,0 +1,46 @@
+"use strict";
+
+function argumentsIsUndefinedString(argumentsArray) {
+  return (
+    argumentsArray.length === 1 &&
+    argumentsArray[0].type === "Literal" &&
+    argumentsArray[0].value === "undefined"
+  );
+}
+
+module.exports = {
+  meta: {
+    schema: [],
+  },
+  create: function(context) {
+    if (context.getFilename().indexOf("packages/babel-plugin-") === -1) {
+      return {};
+    }
+
+    return {
+      CallExpression: function(node) {
+        const callee = node.callee;
+
+        if (
+          callee.type === "MemberExpression" &&
+          argumentsIsUndefinedString(node.arguments)
+        ) {
+          const object = callee.object,
+            property = callee.property;
+
+          if (
+            object.type === "Identifier" &&
+            object.name === "t" &&
+            property.type === "Identifier" &&
+            property.name === "identifier"
+          ) {
+            context.report(
+              node,
+              "prefer path.scope.buildUndefinedNode() to creating an undefined identifier directly"
```

Comment:
hmm, i think the wording i chose here might be strange

perhaps "to" -> "instead of" ?